File: backend/app/schemas/message.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class MessageCreate(BaseModel):
    content: str = Field(min_length=1)
    model: str = "auto"


class MessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    conversation_id: UUID
    role: str
    content: str
    model: str | None
    created_at: datetime


class MessagePairResponse(BaseModel):
    user_message: MessageResponse
    assistant_message: MessageResponse

File: frontend/src/types/message.ts
export interface MessageCreate {
  content: string;
}

export interface Message {
  id: string;
  conversation_id: string;
  role: string;
  content: string;
  created_at: string;
}

export interface MessagePair {
  user_message: Message;
  assistant_message: Message;
}
